packages/cli/core/src/commands/ui/deploy.spec.ts

```diff
@@ -465,6 +473,69 @@ describe('ui:deploy', () => {
         }
       );
 
+    test
+      .stdout()
+      .stderr()
+      .do(() => {
+        doMockConfirm();
+        mockedConfirm.mockResolvedValueOnce(true);
+        mockHostedPageList.mockResolvedValueOnce({
+          items: [{name: validJsonConfig.name, id: 'thisistheid'}],
+        });
+      })
+      .command(['ui:deploy'])
+      .it(
+        'should call hostedPages.list when no page id argument is passed and a page with the same already exists',
+        () => {
+          expect(mockHostedPageList).toHaveBeenCalledWith(
+            expect.objectContaining({
+              filter: validJsonConfig.name,
+              perPage: 1000,
+            })
+          );
+        }
+      );
+
+    test
+      .stdout()
+      .stderr()
+      .do(() => {
+        doMockConfirm();
+        mockedConfirm.mockResolvedValueOnce(true);
+        mockHostedPageList.mockResolvedValueOnce({
+          items: [{name: validJsonConfig.name, id: 'thisistheid'}],
+        });
+      })
+      .command(['ui:deploy'])
+      .it(
+        'should call hostedPages.update when no page id argument is passed, a page with the same already exists, and the user confirms the overwrite',
+        () => {
+          expect(mockHostedPageUpdate).toHaveBeenCalledWith({
```

Comment:
could we also check that `mockHostedPageList` is not called in this case?

Reply:
Is called you mean? But yeah.

Reply:
I don't think this is relevant after 5683cac40da382d7558e717280ef58e15573dacb, but don't hesitate to reopen the conversation